### PR TITLE
lbfgs untargeted attack support

### PR DESCRIPTION
**Pull request by HaojieYuan**
See #1000 for detail description and  code to reproduce error.

### CODE REVIEW

**Comment by npapernot** [comment]
Selecting the second label as a target may not always be the best thing to do in the untargeted setting. Would you be able to instead modify the loss in the untargeted case so that rather than maximizing the score of the target class, the attack minimizes the score of the correct class until the input is misclassified? This would likely be more generally applicable

**Comment by npapernot** [comment]
@HaojieYuan did you get a chance to look at this?

**Comment by HaojieYuan** [comment]
@npapernot  Sorry for the delay, I've been working on other stuff recently.
I will try to figure it out next weekend.

**Comment by HaojieYuan** [comment]
@npapernot  
Can I have my name on the Authors list in `README.md` if this PR is merged?
I think I've already satisfied the requirments:
```
The following authors contributed 100 lines or more (ordered according to the GitHub contributors page)
```
:laughing: it will be super cool for me~
HaojieYuan (University of Science and Technology of China)

File: cleverhans/attacks/lbfgs.py [hunk 1]
```diff
@@ -151,7 +155,12 @@ def __init__(self, sess, x, logits, targeted_label,
         labels=self.targeted_label, logits=self.logits)
     self.l2dist = reduce_sum(tf.square(self.x - self.ori_img))
     # small self.const will result small adversarial perturbation
-    self.loss = reduce_sum(self.score * self.const) + self.l2dist
+    # targeted attack aim at minimize loss against target label
+    # untargeted attack aim at maxmize loss against True label
```
Comment (npapernot) [inline cleverhans/attacks/lbfgs.py]: typo in comment

File: cleverhans/attacks/lbfgs.py [hunk 2]
```diff
@@ -179,6 +188,13 @@ def lbfgs_objective(adv_x, self, targets, oimgs, CONST):
           })
       return loss, grad.flatten().astype(float)
 
+    def attack_success(out, target, targeted_attack):
+      """ returns attack result flag """
```
Comment (npapernot) [inline cleverhans/attacks/lbfgs.py]: could you remove the word "flag"? it is confusing because of tensorflow flags
